src/GoogleAuthManager.php

```diff
@@ -3,93 +3,193 @@
 class GoogleAuthManager extends OAuth2Manager {
 
   /**
-   * The request object.
+   * The logger channel.
    *
-   * @var \Symfony\Component\HttpFoundation\Request
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
    */
-  protected $request;
+  protected $loggerFactory;
 
   /**
-   * The Google service client.
+   * The event dispatcher.
    *
-   * @var \Google_Client
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  protected $eventDispatcher;
+
+  /**
+   * The entity field manager.
+   *
+   * @var \Drupal\Core\Entity\EntityFieldManagerInterface
+   */
+  protected $entityFieldManager;
+
+  /**
+   * The url generator.
+   *
+   * @var \Drupal\Core\Routing\UrlGeneratorInterface
+   */
+  protected $urlGenerator;
+
+  /**
+   * The Google client object.
+   *
+   * @var \League\OAuth2\Client\Provider\Google
    */
   protected $client;
+  /**
+   * The Google access token.
+   *
+   * @var League\OAuth2\Client\Token\AccessToken
+   */
+  protected $token;
 
   /**
-   * Code returned by Google for authentication.
+   * The Google user.
    *
-   * @var string
+   * @var \League\OAuth2\Client\Provider\GoogleUser
    */
-  protected $code;
+  protected $user;
 
   /**
-   * GoogleLoginManager constructor.
+   * Constructor.
    *
-   * @param \Symfony\Component\HttpFoundation\RequestStack $request
-   *   Used to get the parameter code returned by Google.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   Used for logging errors.
+   * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $event_dispatcher
+   *   Used for dispatching events to other modules.
+   * @param \Drupal\Core\Entity\EntityFieldManagerInterface $entity_field_manager
+   *   Used for accessing Drupal user picture preferences.
+   * @param \Drupal\Core\Routing\UrlGeneratorInterface $url_generator
+   *   Used for generating absoulute URLs.
    */
-  public function __construct(RequestStack $request) {
-    $this->request = $request->getCurrentRequest();
+  public function __construct(LoggerChannelFactoryInterface $logger_factory, EventDispatcherInterface $event_dispatcher, EntityFieldManagerInterface $entity_field_manager, UrlGeneratorInterface $url_generator) {
+    $this->loggerFactory      = $logger_factory;
+    $this->eventDispatcher    = $event_dispatcher;
+    $this->entityFieldManager = $entity_field_manager;
+    $this->urlGenerator       = $url_generator;
   }
 
   /**
-   * {@inheritdoc}
+   * Authenticates the users by using the access token.
+   *
+   * @return $this
+   *   The current object.
    */
   public function authenticate() {
-    $this->client->setAccessToken($this->getAccessToken());
-    return $this;
+    $this->token = $this->client->getAccessToken('authorization_code',
+      ['code' => $_GET['code']]);
+
+    return $this->token;
   }
 
   /**
-   * {@inheritdoc}
+   * Gets the data by using the access token returned.
+   *
+   * @return array
```

Comment:
You're not returning an array in getUserInfo(), you're returning a \League\OAuth2\Client\Provider\GoogleUser object